src/olympia/devhub/views.py

```diff
@@ -677,7 +677,7 @@ def file_validation(request, addon_id, addon, file_id):
 
 
 @csrf_exempt
-@dev_required(allow_reviewers_for_read=True)
+@dev_required(allow_reviewers_for_read=True, qs=Addon.unfiltered.all)
```

Comment:
So, following the implications of that wasn't trivial. It looks like with that change, for deleted add-ons:
- developers still don't have access because `check_addon_ownership()` which always return `False` for deleted add-ons
- reviewers retain read-only access (the only thing required for this view) because they bypass that same check entirely, thanks to `allow_reviewers_for_read=True`.

Can you add a test covering the fact that developers don't get access, if there isn't already one ? Also, add a comment explaining who gets access and why next to that decorator call would be helpful.

Reply:
I've addressed these issues in the latest commit.